options.js

```diff
@@ -69,24 +73,41 @@ function restoreOptions(callback) {
 
 // Saves options to chrome.storage
 function saveOptions() {
-    let needsReload = false;
     try {
-        let options = JSON.parse(document.getElementById("optionsTextArea").value);
+        const optionsStr = document.getElementById("optionsTextArea").value;
+        let options = JSON.parse(optionsStr);
         if (options.lateScoreAdjustment > 0) {
             alert("Late score adjustment has to be negative. Defaulting the value to 0.");
             options.lateScoreAdjustment = 0;
-            needsReload = true;
         }
+
+        // If firstStudent isn't a valid field, trim will produce undefined (falsey)
+        // so the field will default to "a". Conveniently, this also means firstStudent
+        // can't be the empty string since that's also falsey.
+        options.firstStudent = options.firstStudent?.trim() || "a";
+        // To include directory IDs that start with z, use ASCII codepoint directly
+        // after z as the default upper bound
+        options.lastStudent = options.lastStudent?.trim() || "{";
+        if (options.firstStudent > options.lastStudent) {
+            const tmp = options.firstStudent;
+            options.firstStudent = options.lastStudent;
+            options.lastStudent = tmp;
+        }
+
         chrome.storage.sync.set(
-            options
-            , function () {
+            options,
+            function () {
                 // Update status to let user know options were saved.
                 let status = document.getElementById('status');
                 status.textContent = 'Options saved.';
                 setTimeout(function () {
                     status.textContent = '';
                 }, 750);
             });
+
+        if (optionsStr !== JSON.stringify(options)) {
+            restoreOptionsLocal();
+        }
     } catch (error) {
```

Comment:
Instead of using flag to control options reload, compare previous and updated string representation (perhaps a bit clunky)

Reply:
I think this is fine for now.